### ISSUE TITLE
Can't split a SortedByQueryHits with an unordered factor

### ISSUE THREAD

**Issue by LTLA** [Bioconductor/S4Vectors#39]
```r
library(S4Vectors)

split(Hits(c(1,1,2,2), c(1,2,3,4), 2, 4), c(1,1,2,2))
## HitsList of length 2
## names(2): 1 2

split(Hits(c(1,1,2,2), c(1,2,3,4), 2, 4, sort.by.query=TRUE), c(1,1,2,2))
## SortedByQueryHitsList of length 2
## names(2): 1 2

split(Hits(c(1,1,2,2), c(1,2,3,4), 2, 4), c(1,2,1,2))
## HitsList of length 2
## names(2): 1 2

split(Hits(c(1,1,2,2), c(1,2,3,4), 2, 4, sort.by.query=TRUE), c(1,2,1,2))
## Error in extractROWS(x, idx) : 
##   Problem(s) found when testing validity of SortedByQueryHits object
##   returned by subsetting operation: 'queryHits(x)' must be sorted. Make
##   sure to use a subscript that results in a valid SortedByQueryHits
##   object.
```

<details>
<summary>Session information</summary>

```
R version 3.6.0 Patched (2019-05-02 r76458)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.2 LTS

Matrix products: default
BLAS:   /home/luna/Software/R/R-3-6-branch-dev/lib/libRblas.so
LAPACK: /home/luna/Software/R/R-3-6-branch-dev/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
[1] S4Vectors_0.23.3    BiocGenerics_0.31.2

loaded via a namespace (and not attached):
[1] compiler_3.6.0 IRanges_2.19.5
```
</details>

**Comment by hpages** [comment]
Thanks Aaron. Should be addressed in S4Vectors 0.23.10 (see commit bf02e52b577b773b7b1723b3ec936b74bf53ab45). Earlier I also moved the `splitAsList` generic from IRanges to S4Vectors so make sure to also install the latest IRanges (2.19.7).

**Comment by LTLA** [comment]
Looks good, thanks.
